### PR TITLE
[MAINTENANCE] py311-min-versions ci test

### PR DESCRIPTION
**Pull request by Kilo59**
Add a min-version test for Python 3.11.
Move min version tests to a matrix.

## TODO
- [ ] update the Github required status checks
- [x] use `invoke ci-test`

---------

- [x] Description of PR changes above includes a link to [an existing GitHub issue](https://github.com/great-expectations/great_expectations/issues)
- [x] PR title is prefixed with one of: [BUGFIX], [FEATURE], [DOCS], [MAINTENANCE], [CONTRIB]
- [x] Code is linted - run `invoke lint` (uses `black` + `ruff`)
- [x] Appropriate tests and docs have been updated

For more details, see our [Contribution Checklist](https://docs.greatexpectations.io/docs/contributing/contributing_checklist), [Coding style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/code_style), and [Documentation style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/docs_style).

After you submit your PR, keep the page open and **monitor the statuses of the various checks made by our continuous integration process at the bottom of the page. Please fix any issues that come up** and [reach out on Slack](https://greatexpectations.io/slack) if you need help. Thanks for contributing!

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -245,60 +245,24 @@ jobs:
...
-          cache: "pip"
-          cache-dependency-path: |
-            reqs/requirements-dev-test.txt
-      - name: Install dependencies
-        run: pip install . -c ci/constraints-test/py39-min-install.txt -r reqs/requirements-dev-test.txt
-      - name: Run the tests
-        run: invoke tests --unit
-
-  py310-min-versions:
+  min-versions:
```
Comment (billdirks) [inline .github/workflows/ci.yml:333]: I like the consolidation of this into 1 test stage.

I notice for this we use `invoke test` but for all stages we use `invoke ci-test`. Can you update that here?

Comment (Kilo59) [reply .github/workflows/ci.yml:333]: Done

File: ci/constraints-test/py310-min-install.txt [hunk 2]
```diff
@@ -1,3 +1,4 @@
 numpy==1.22.4
+# min version for py310 is pandas 1.3.0 but wheel is no longer published
```
Comment (Shinnnyshinshin) [inline ci/constraints-test/py310-min-install.txt:2]: very helpful. Thank you 🙇🏼

**Review by Shinnnyshinshin — APPROVED** [review]
LGTM!

**Comment by Kilo59** [comment]
Closing for now, will revisit

**Pull request closed without merging**